### ISSUE TITLE
Intel only - Model is empty on rollover of ComputerInfo button

### ISSUE THREAD

**Issue by johngsea** [root3nl/SupportApp#134]
possible bug...

macOS 13.4.1 (c) 
MacBook Pro 16-inch, 2019
builds: latest master and development

<img width="384" alt="empty-model-name-Intel" src="https://github.com/root3nl/SupportApp/assets/807877/33041ac9-40ab-48cc-a088-4e2e2f888fe3">

```
src/Support/Views/ButtonViews/ComputerNameSubview.swift:

a fix, but not an ideal one:
let secondSubtitleString = computerinfo.modelNameString.isEmpty ? computerinfo.modelShortName :  computerinfo.modelNameString

 ItemDouble(title: NSLocalizedString("Computer Name", comment: ""), secondTitle: NSLocalizedString("Model", comment: ""), subtitle: computerinfo.hostname, secondSubtitle: secondSubtitleString, linkType: aboutLinkType, link: aboutLink, image: computerinfo.computerNameIcon, symbolColor: Color(NSColor(hex: "\(customColor)") ?? NSColor.controlAccentColor), hoverEffectEnable: true)
```

That fix gives the correct output on both Silicon and Intel:
<img width="378" alt="fixed-Intel" src="https://github.com/root3nl/SupportApp/assets/807877/bb47d855-6e3c-43e1-8128-566e222bede7">

But the underlying issue still remains in that the call to Apple's url returns no results. Tested on several machines. 

also verified using curl that the result is empty:
```curl "https://support-sp.apple.com/sp/product?cc=[last_4_serial_number]"```
```<?xml version="1.0" encoding="utf-8" ?><root><name>CPU Name</name><configCode></configCode><locale>en_US</locale></root>%  ```

2023-07-19 10:06:36.608618-0700 Support[11230:1383308] [ComputerInfo] Error matching serial number with model and introduction year...
2023-07-19 10:06:36.608670-0700 Support[11230:1383308] [ComputerInfo] Full Model Name:
2023-07-19 10:06:36.608688-0700 Support[11230:1383308] [ComputerInfo] Model Year:

possible fix?
system_profiler SPHardwareDataType

```
...Hardware:

    Hardware Overview:

      Model Name: MacBook Pro
      Model Identifier: MacBookPro16,1
      Processor Name: 8-Core Intel Core i9
```

Thanks!

**Comment by jordywitteman** [comment]
@johngsea thanks for bringing this up. It seems like Apple is no longer returning the model name. I also tried a couple of Intel serial numbers and none is working. I think your suggestion to go with the model short name is the best option as it is probably not possible anymore to get the full name on Intel. Also, troubleshooting on Intel gets harder as Apple Silicon is taking over.

**Comment by johngsea** [comment]
Thank you, Jordy! We'll use the short name. Agree, Intel is being phased out anyway.

**Comment by wakco** [comment]
The link: https://support-sp.apple.com/sp/product?cc=[last_4_serial_number] intel Macs and limited Apple Silicon M1 (Pro/Max/Ultra excluded) Macs only. This is because Apple has switched to a fully randomised Serial Number scheme, therefore no longer adhering to the legacy Serial Number format that included model codes, year and week in the last decade it was made, where it was manufactured, and one or two more bits of information beyond a basic count off the assembly line. As a result the web-link has been an unreliable source of information for 2-3 years now.

**Comment by jordywitteman** [comment]
This was partially resolved in v2.5. On Intel, it now shows the model shortname because it seems it is no longer possible to obtain the full model name from the Apple API endpoint [https://support-sp.apple.com/sp/product?cc=[last_4_serial_number]](https://support-sp.apple.com/sp/product?cc=%5Blast_4_serial_number%5D)

**Comment by PicoMitchell** [comment]
You can still get the full Marketing Model Name for Intel Macs from the backend URL API of the [Specs Search](https://support.apple.com/specs) Apple Support page. (At least you can for now, who knows if Apple will break this one day too.)

Here's an example of how to format the URL and parse the returned JSON: https://github.com/freegeek-pdx/macOS-Testing-and-Deployment-Scripts/blob/main/Other%20Scripts/get_specs_url_from_serial.sh

That code does 2 URL calls to get the Tech Specs URL, but all you need is the first one just to get the Marketing Model Name.

TLDR, call `https://km.support.apple.com/kb/index?page=categorydata&serialnumber=[SERIALNUMBER]` and then get the the value from the `name` key in the returned JSON.

**Comment by PicoMitchell** [comment]
Also, serendipitously, not long before the `support-sp` URL API went down, I brute force saved every single config code with their Marketing Model Name: https://github.com/freegeek-pdx/macOS-Testing-and-Deployment-Scripts/blob/main/Other%20Scripts/every_apple_serial_config_code_with_marketing_model_name.txt

I also processed all of that down to just the Intel Mac Marketing Model Name with the relevant Config Codes and Model IDs (only Model IDs are listed in this file for models that don't have multiple possible Marketing Model Name): https://github.com/freegeek-pdx/macOS-Testing-and-Deployment-Scripts/blob/main/Other%20Scripts/every_intel_mac_marketing_model_name_with_grouped_model_ids_and_serial_config_codes.txt

And here's an example of how that text can be processed to get an Intel Macs Marketing Model Name totally offline: https://github.com/freegeek-pdx/macOS-Testing-and-Deployment-Scripts/blob/3e5f7b24d8efc585de76932ef19dd61f4e530a42/fgMIB%20Resources/fg-install-os.sh#L223-L389

That code is `bash`, but the same concept could be done in any language, and the data could be converted into a native data structure rather than just processing a text list, if desired.
